### ISSUE TITLE
Strange behaviour

### ISSUE THREAD

**Issue by LiamGoodacre** [23Skidoo/pointful#8]
Strange behaviour discovered using: [http://pointfree.info/#input=...](http://pointfree.info/#input=(\\l%20r%20t%20f%20->%20l%20(r%20t%20f)%20f)%20(\\t%20f%20->%20t)%20(\\t%20f%20->%20f)%201%200)

~~~haskell
-- correct: `and true false` is false
> (\l r t f -> l (r t f) f) (\t f -> t) (\t f -> f)
(\ t f -> f)

-- incorrect: `and true false 1` should be `\f -> f`
> (\l r t f -> l (r t f) f) (\t f -> t) (\t f -> f) 1
(\ f -> 1)

-- incorrect: `and true false 1 0` should be `0`
> (\l r t f -> l (r t f) f) (\t f -> t) (\t f -> f) 1 0
1
~~~

**Comment by 23Skidoo** [comment]
Does Lambdabot's `@unpl` have the same bug? If not, I'll update this copy of the code.

**Comment by tfausak** [comment]
Lambdabot has the same bug.

```
[09:23] <fozworth> @unpl (\l r t f -> l (r t f) f) (\t f -> t) (\t f -> f) 1 0
[09:23] <lambdabot> 1
```

**Comment by 23Skidoo** [comment]
Then the issue should be moved to their bug tracker.

**Comment by 23Skidoo** [comment]
Closing in favour of https://github.com/lambdabot/lambdabot/issues/142.

**Comment by 23Skidoo** [comment]
Fixed in pointful-1.0.8.
